monkey/monkey_island/cc/ui/src/components/ui-components/AdvancedMultiSelect.js

```diff
@@ -41,81 +32,168 @@ function getObjectFromRegistryByRef(refString, registry) {
-  getDefaultPaneParams(schema.items.$ref, registry);
-  const selectValue = cloneDeep(value);
-  return (
-    <div className={'advanced-multi-select'}>
-      <Card.Header>
-        <Button key={`${props.schema.title}-button`} value={value}
+class AdvancedMultiSelect extends React.Component {
+  constructor(props) {
+    super(props)
+
+    this.state = {
+      masterCheckboxState: this.getMasterCheckboxState(props.value),
+      infoPaneParams: getDefaultPaneParams(props.schema.items.$ref, props.registry)
+    };
+
+    this.onMasterCheckboxClick = this.onMasterCheckboxClick.bind(this);
+    this.onChildCheckboxClick = this.onChildCheckboxClick.bind(this);
+    this.setPaneInfo = this.setPaneInfo.bind(this, props.schema.items.$ref, props.registry);
+  }
+
+  onMasterCheckboxClick() {
+    let newValues = this.props.options.enumOptions.map(({value}) => value);
+
+    if (this.state.masterCheckboxState == MasterCheckboxState.ALL) {
+      newValues = [];
+    }
+
+    this.props.onChange(newValues);
+    this.setMasterCheckboxState(newValues);
+  }
+
+  onChildCheckboxClick(value) {
+    let selectValues = this.getSelectValuesAfterClick(value)
+    this.props.onChange(selectValues);
+
+    this.setMasterCheckboxState(selectValues);
+  }
+
+  getSelectValuesAfterClick(clickedValue) {
+    const valueArray = cloneDeep(this.props.value);
+
+    if (valueArray.includes(clickedValue)) {
+      return valueArray.filter(e => e !== clickedValue);
+    } else {
+      valueArray.push(clickedValue);
+      return valueArray;
+    }
+  }
+
+  setMasterCheckboxState(selectValues) {
+    this.setState(() => ({
+      masterCheckboxState: this.getMasterCheckboxState(selectValues)
+    }));
+  }
+
+  getMasterCheckboxState(selectValues) {
+    if (selectValues.length == 0) {
+      return MasterCheckboxState.NONE;
+    }
+
+    if (selectValues.length != this.props.options.enumOptions.length) {
+      return MasterCheckboxState.MIXED;
+    }
+
+    return MasterCheckboxState.ALL;
+  }
+
+  setPaneInfo(refString, registry, itemKey) {
+    let definitionObj = getFullDefinitionByKey(refString, registry, itemKey);
+    this.setState({infoPaneParams: {title: definitionObj.title, content: definitionObj.info, link: definitionObj.link}});
+  }
+
+  render() {
+    const {
+      schema,
+      id,
+      options,
+      value,
+      required,
+      disabled,
+      readonly,
+      multiple,
+      autofocus
+    } = this.props;
+
+    const {enumOptions} = options;
+
+    return (
+      <div className={'advanced-multi-select'}>
+        <MasterCheckbox title={schema.title} value={value}
+          disabled={disabled} onClick={this.onMasterCheckboxClick}
+          checkboxState={this.state.masterCheckboxState}/>
+        <Form.Group
+          style={{height: `${getComponentHeight(enumOptions.length)}px`}}
+          id={id} multiple={multiple} className='choice-block form-control'
```

Comment:
I wonder if `className` actually works, or do we need to replace `'` with `

Reply:
In my browser it's rendered as 
`<div id="root_exploiters_exploiter_classes" class="choice-block form-control form-group" ...`, so I think it's ok as is.